Specs/Scene/ModelExperimental/ModelExperimentalSpec.js

```diff
@@ -1210,6 +1272,259 @@ describe(
+        expect(model.heightReference).toEqual(HeightReference.CLAMP_TO_GROUND);
+        expect(sceneWithMockGlobe.globe.callback).toBeDefined();
+      });
+    });
+
+    it("creates height update callback after setting height reference", function () {
+      return loadAndZoomToModelExperimental(
+        {
+          gltf: boxTexturedGltfUrl,
+          modelMatrix: Transforms.eastNorthUpToFixedFrame(
+            Cartesian3.fromDegrees(-72.0, 40.0)
+          ),
+          heightReference: HeightReference.NONE,
+          scene: sceneWithMockGlobe,
+        },
+        sceneWithMockGlobe
+      ).then(function (model) {
+        expect(model.heightReference).toEqual(HeightReference.NONE);
+        expect(sceneWithMockGlobe.globe.callback).toBeUndefined();
+
+        model.heightReference = HeightReference.CLAMP_TO_GROUND;
+        expect(model.heightReference).toEqual(HeightReference.CLAMP_TO_GROUND);
+        sceneWithMockGlobe.renderForSpecs();
+        expect(sceneWithMockGlobe.globe.callback).toBeDefined();
+      });
+    });
+
+    it("updates height reference callback when the height reference changes", function () {
+      return loadAndZoomToModelExperimental(
+        {
+          gltf: boxTexturedGltfUrl,
+          modelMatrix: Transforms.eastNorthUpToFixedFrame(
+            Cartesian3.fromDegrees(-72.0, 40.0)
+          ),
+          heightReference: HeightReference.CLAMP_TO_GROUND,
+          scene: sceneWithMockGlobe,
+        },
+        sceneWithMockGlobe
+      ).then(function (model) {
+        expect(sceneWithMockGlobe.globe.callback).toBeDefined();
+
+        model.heightReference = HeightReference.RELATIVE_TO_GROUND;
+        sceneWithMockGlobe.renderForSpecs();
+        expect(sceneWithMockGlobe.globe.removedCallback).toEqual(true);
+        expect(sceneWithMockGlobe.globe.callback).toBeDefined();
+
+        sceneWithMockGlobe.globe.removedCallback = false;
+        model.heightReference = HeightReference.NONE;
+        sceneWithMockGlobe.renderForSpecs();
+        expect(sceneWithMockGlobe.globe.removedCallback).toEqual(true);
+        expect(sceneWithMockGlobe.globe.callback).toBeUndefined();
+      });
+    });
+
+    it("updates height reference callback when the model matrix changes", function () {
+      return loadAndZoomToModelExperimental(
+        {
+          gltf: boxTexturedGltfUrl,
+          modelMatrix: Transforms.eastNorthUpToFixedFrame(
+            Cartesian3.fromDegrees(-72.0, 40.0)
+          ),
+          heightReference: HeightReference.CLAMP_TO_GROUND,
+          scene: sceneWithMockGlobe,
+        },
+        sceneWithMockGlobe
+      ).then(function (model) {
+        expect(sceneWithMockGlobe.globe.callback).toBeDefined();
+
+        const matrix = Matrix4.clone(model.modelMatrix);
+        const position = Cartesian3.fromDegrees(-73.0, 40.0);
+        matrix[12] = position.x;
+        matrix[13] = position.y;
+        matrix[14] = position.z;
+
+        model.modelMatrix = matrix;
```

Comment:
Would this still update if the model matrix was updated in place? if so, maybe add that to this test?